Shopify/eng-intern-assessment-rails
# Aastha Sharma - Backend Technical Challenge (eng-intern-assesment-rails)

Summary:
This pull request completes the application of Encyclopedia, including the CRUD actions and the search feature. The entire application also follows the MVC principles and rails conventions. The tests given for the Article model all pass successfully. 

Changes:
- The Gemfile was changed to run ruby version 3.2.2
- The Article Model (with a search method) and Database were created.
- The Article's controller was created with CRUD + search methods, also linking to the views
- The views were created for each CRUD + search page
- The Router was modified to generate URLs for each page, creating an application.

How to Test:
- Make sure the database exists for Articles
- Run the tests on the test/models/article_test.rb
- Run the application locally on the rails server.
- Check out the CRUD and the search features!
